jepsen/scalardb/src/scalardb/transfer_append.clj

```diff
@@ -0,0 +1,328 @@
+             [policies :refer :all]
+             [query :refer :all]]
+            [clojure.tools.logging :refer [debug info warn]]
+            [clojure.core.reducers :as r]
+            [knossos.op :as op]
+            [scalardb.core :as scalar])
+  (:import (java.util ArrayList)
+           (com.scalar.database.api Consistency
+                                    Get
+                                    Isolation
+                                    Put
+                                    Scan
+                                    Scan$Ordering
+                                    Scan$Ordering$Order
+                                    Result)
+           (com.scalar.database.io IntValue
+                                   Key)
+           (com.scalar.database.exception.transaction CommitException
+                                                      CrudException
+                                                      UnknownTransactionStatusException UncommittedRecordException)
+           (com.scalar.database.transaction.consensuscommit ConsensusCommit)
+           (com.scalar.database.exception.storage NoMutationException)))
+
+(def ^:const KEYSPACE "jepsen")
+(def ^:const TABLE "transfer")
+(def ^:const ACCOUNT_ID "account_id")
+(def ^:const BALANCE "balance")
+(def ^:const AGE "age")
+
+(def ^:const INITIAL_BALANCE 10000)
+(def ^:const NUM_ACCOUNTS 10)
+(def ^:const total-balance (* NUM_ACCOUNTS INITIAL_BALANCE))
+
+(def ^:const NUM_FAILURES_FOR_RECONNECTION 1000)
+
+(defn- create-transfer-table!
+  [session test]
+  (cql/create-keyspace session KEYSPACE
+                       (if-not-exists)
+                       (with {:replication {"class" "SimpleStrategy"
+                                            "replication_factor" (:rf test)}}))
+  (cql/use-keyspace session KEYSPACE)
+  (cql/create-table session TABLE
+                    (if-not-exists)
+                    (column-definitions {:account_id             :int
+                                         :age                    :int
+                                         :balance                :int
+                                         :tx_id                  :text
+                                         :tx_prepared_at         :bigint
+                                         :tx_committed_at        :bigint
+                                         :tx_state               :int
+                                         :tx_version             :int
+                                         :before_balance         :int
+                                         :before_tx_committed_at :bigint
+                                         :before_tx_id           :text
+                                         :before_tx_prepared_at  :bigint
+                                         :before_tx_state        :int
+                                         :before_tx_version      :int
+                                         :primary-key            [:account_id :age]}))
```

Comment:
Just a confirmation. It means using `age` as a clustering key ?

Reply:
Yes. Thank you.